ng/js/controllers/new_posting.js

```diff
@@ -0,0 +1,270 @@
+      }
+      if (!empty && !text.match(/^[\w ]*[^\W_][\w ]*$/)) {
+        $scope.titleValidated = false;
+        $scope.titleErrorMessages.push('Title is not alphanumeric');
+      }
+    };
+
+    // Validate each price field. Gets called in 'focusableForm'
+    // directive, based on which input field was last 'blurred',
+    // or unfocused.
+    $scope.validatePrices = function () {
+      var lowPrice = $scope.newPosting.price.minimum;
+      var highPrice = $scope.newPosting.price.maximum;
+      var fLow = parseFloat(lowPrice.replace(/,/g, ''));
+      var fHigh = parseFloat(highPrice.replace(/,/g, ''));
+      var msg1 = 'Either the minimum or maximum price (or both if posting a price range) is required';
+      var msg2 = 'Minimum price is invalid';
+      var msg3 = 'Maximum price is invalid';
+      var regex = /^([1-9]\d{0,2}(,\d{3})*|([1-9]\d*))(\.\d{2})?$/;
+      $scope.priceErrorMessages = [];
+      $scope.pricesValidated = true;
+      // If min price is empty
+      if (lowPrice === '') {
+        // If high price is also empty, mark invalid
+        if (highPrice === '') {
+          $scope.pricesValidated = false;
+          if ($scope.priceErrorMessages.indexOf(msg1) === -1) {
+            $scope.priceErrorMessages.push(msg1);
+          }
+        }
+      } else if (!lowPrice.match(regex)) { // If low price isnt empty, but has invalid input
+        $scope.pricesValidated = false;
+        if (lowPrice.includes('$')) {
+          $scope.priceErrorMessages.push('Do not include a dollar sign in minimum price');
+        } else {
+          $scope.priceErrorMessages.push(msg2);
+        }
+      }
+      // If max price is empty along with min price (will not repeat first 'if' since its in an if/else if block)
+      if (highPrice === '') {
+        if (lowPrice === '') {
+          $scope.pricesValidated = false;
+          if ($scope.priceErrorMessages.indexOf(msg1) === -1) {
+            $scope.priceErrorMessages.push(msg1);
+          }
+        }
+      } else if (!highPrice.match(regex)) {
+        $scope.pricesValidated = false;
+        if (highPrice.includes('$')) {
+          $scope.priceErrorMessages.push('Do not include a dollar sign in maximum price');
+        } else {
+          $scope.priceErrorMessages.push(msg3);
+        }
+      }
+      // Min is equal to or greater than max price
+      if (fLow >= fHigh) {
+        $scope.pricesValidated = false;
+        // Don't want to include redundant error messages. 
+        // If its not already saying its invalid, then include this message
+        if (!$scope.priceErrorMessages.includes(msg2) && !$scope.priceErrorMessages.includes(msg3)) {
+          $scope.priceErrorMessages.push('Minimum price must be less than the maximum price');
+        }
+      }
+    };
+  })
+
+  // This directive is added to the form in createPost.html, used to execute validation
+  // during the on-blur events.
+  .directive('focusableForm', function ($timeout, $animate) {
```

Comment:
Annotate in angular refers to explicitly defining the providers being injected: `['$scope', function($scope){...}] `. Makes it safe to minify the code